api/namex/resources/name_requests/name_request.py

```diff
@@ -250,72 +259,152 @@ def handle_patch_edit(self, nr_model):
         nr_model = self.update_nr_fields(nr_model, nr_model.stateCd)
 
         # This handles the updates for NRO and Solr, if necessary
-        # TODO: Do we update network services?
         self.update_records_in_network_services(nr_model)
         return nr_model
 
     def handle_patch_upgrade(self, nr_model):
-        # TODO: Only works for DRAFT state
+        """
+        Upgrade the Name Request to priority, create the payment and save the record.
+        :param nr_model:
+        :return:
+        """
+        if not nr_model.stateCd == State.DRAFT:
+            raise NameRequestException(message='Error upgrading Name Request, request is in an invalid state!')
 
         # This handles updates if the NR state is 'patchable'
         nr_model = self.update_nr_fields(nr_model, nr_model.stateCd)
 
-        # TODO: Update Oracle (change_nr?) and payment related stuff
-        # This will generate a new payment Id and then the NR will have two payments.
-        # We have not accounted for multiple payments.
-        # We will need to add a request_payment model (request_id and payment_id)
-        # This handles the updates for NRO and Solr, if necessary
-        self.update_records_in_network_services(nr_model)
-        return nr_model
+        # TODO: Any other supported types? What role does frontend have in this?
+        #  Also, this will generate a new payment Id and then the NR will have two payments.
+        nr_name = nr_model.names[0]
+        nr_applicant = nr_model.applicants[0]
+
+        if nr_name and nr_applicant:
+            payment_request = {
+                'paymentInfo': {
+                    'methodOfPayment': 'CC',
+                },
+                'filingInfo': {
+                    'date': date.today(),
+                    'filingTypes': [
+                        {
+                            'filingDescription': 'NM620: ' + nr_name.name,
```

Comment:
Please set the filing description to null so the receipt does not have the fee code NM620 and also so the receipt does not have the name as a header.  This in on advice from Sumesh.

Reply:
Noted.